### PR TITLE
DB-11129 Implement POSSTR function

### PR DESCRIPTION
**Pull request by msirek**
Implement the POSSTR function, which finds the position of a search string in an expression.
Example:

> values posstr('abc','bc');
> 1          
> //-----------
> 2

### CODE REVIEW

**Comment by msirek** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

File: db-engine/src/main/java/com/splicemachine/db/iapi/types/StringDataValue.java [hunk 1]
```diff
@@ -204,6 +204,15 @@ StringDataValue upper(StringDataValue result)
 	StringDataValue upperWithLocale(StringDataValue str, StringDataValue locale, StringDataValue result)
 							throws StandardException;
 
+	/**
+	 * Find the position of the searchString in an expressionString
+	 * @param expressionString The string to search
+	 * @param searchString  The substring to search for
+	 * @return 0 if not found, otherwise the 1-based position of the first character in the match
```
Comment (hatyo) [inline db-engine/src/main/java/com/splicemachine/db/iapi/types/StringDataValue.java:211]: .. or null, if either operand is null.

**Review by hatyo — APPROVED** [review]

File: splice_machine/src/test/java/com/splicemachine/derby/utils/SpliceStringFunctionsIT.java [hunk 2]
```diff
@@ -464,6 +464,77 @@ public void testInitcapFunction() throws Exception {
         }
     }
 
+    @Test
```
Comment (arnaud-splice) [inline splice_machine/src/test/java/com/splicemachine/derby/utils/SpliceStringFunctionsIT.java:467]: Can you add a test with useSpark=true and with/without useNativeSpark=true?

**Review by arnaud-splice — CHANGES_REQUESTED** [review]
Great change! One request for an additional test

File: db-engine/src/main/java/com/splicemachine/db/iapi/types/StringDataValue.java (hunk 1 above)
Comment (msirek) [reply db-engine/src/main/java/com/splicemachine/db/iapi/types/StringDataValue.java:211]: Fixed

**Comment by msirek** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

File: splice_machine/src/test/java/com/splicemachine/derby/utils/SpliceStringFunctionsIT.java (hunk 2 above)
Comment (msirek) [reply splice_machine/src/test/java/com/splicemachine/derby/utils/SpliceStringFunctionsIT.java:467]: Added tests which should automatically pick native spark for broadcast and sortmerge join, if eligible.

**Comment by msirek** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/2462/

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

**Review by arnaud-splice — APPROVED** [review]

**Pull request merged**